### PR TITLE
Implement zoom and rotate for preview app

### PR DESCRIPTION
**Pull request by JanAckermann**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/7160

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublic2</strong> https://drone.owncloud.com/owncloud/web/29892/37/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicExpire-shareByPublicLinkExpiringLinks_feature-L62.png
  


</p></details>
:boom: The oC10SharingPublic2 tests pipeline failed. The build has been cancelled.

**Comment by JanAckermann** [comment]
I really don't know what to "unit-test" here

File: packages/web-app-preview/src/App.vue [hunk 1]
```diff
@@ -350,13 +439,32 @@ export default defineComponent({
       }
       this.activeIndex--
       this.updateLocalHistory()
+    },
+    imageShrink() {
+      this.currentImageZoom = Math.max(0.1, Math.round(this.currentImageZoom * 0.8 * 20) / 20)
```
Comment (kulmann) [inline packages/web-app-preview/src/App.vue:444]: Please introduce a method similar to the following:

```js
calculateZoom(zoom: number, factor: number) {
      return Math.round(zoom * factor * 20) / 20
    }
```

and use it in all the places where you have the respective Math.round statement at the moment.

**Review by kulmann — CHANGES_REQUESTED** [review]
The controls for zoom feel a bit jumpy, depending on the percentage value shown in the controls. Could you make this a fixed width somehow?
Other than that only a nitpick about some code deduplication. Feature itself works like a charm, I love it!

I was a bit irritated that the zoom and rotate controls only appear after the image is loaded (that looks funny when browsing through a folder of only images, because the zoom and rotate controls disappear and then reappear all the time). But that will be somewhat fixed when we introduce preloading of images. So nothing we need to worry about here.

**Comment by JanAckermann** [comment]
Thx for the feedback!

**Review by kulmann — APPROVED** [review]
Love it 😍

**Pull request merged**